components/base/kinematicbase/execution.go

```diff
@@ -481,6 +480,7 @@ func (ptgk *ptgBaseKinematics) courseCorrect(
 			if solution.stepIdx < len(arcSteps)-1 {
 				newArcSteps = append(newArcSteps, arcSteps[solution.stepIdx+1:]...)
 			}
+			ptgk.inputLock.Unlock()
```

Comment:
> course correction of a kinematic base is mutating the map that is being passed via pointer with ExecutionState

@biotinker I believe this should solve the issue.

Reply:
I do not think so. `newArcSteps` is a new variable; we are not writing anything in the newly-protected 10 lines that would affect something previously passed out.

It is a *write* in course correction that affects a *read* in check plan.